### ISSUE TITLE
Nullability of class fields are not inherited on assignment

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#31395]
A warning is reported for `d.F`. (Compare with `t.F`.)
```c#
#nullable enable

class C
{
    internal object? F;
}

struct S
{
    internal object? F;
}

class Program
{
    static void F1()
    {
        var c = new C() { F = "" };
        F(c.F); // ok
        var d = c;
        F(d.F); // warning
    }

    static void F2()
    {
        var s = new S() { F = "" };
        F(s.F); // ok
        var t = s;
        F(t.F); // ok
    }

    static void F(object o)
    {
    }
}
```
